packages/core/src/button/Button.tsx

```diff
@@ -72,15 +89,41 @@ export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
           withBaseName(variant),
           {
             [withBaseName("disabled")]: disabled,
+            [withBaseName(`loading-${variant}`)]: loading,
```

Comment:
We already have a variant class, why do we need to add loading-(variant) classes?

Reply:
we need it, for handling css part, of different variant, like border for different variants, while loiading